### DISCUSSION TITLE
What is the best way to have the entries of Scope.java in semantic-model ?

### DISCUSSION THREAD

**Discussion by dulajdilshan** [ballerina-platform/ballerina-lang#27968] [Q&A]
Hi,

The current implementation is to have this as a Map https://github.com/ballerina-platform/ballerina-lang/blob/eefa8da5eaa5424b1804666711e4bceef010a702/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/Scope.java#L37

But, when we have two symbols with the same name (same key), we override the entry with new value 
https://github.com/ballerina-platform/ballerina-lang/blob/eefa8da5eaa5424b1804666711e4bceef010a702/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/Scope.java#L49
in the `define` function
```java
public void define(Name name, BSymbol symbol) {
        ScopeEntry current = entries.get(name);
        if (current == null) {
            current = NOT_FOUND_ENTRY;
        }

        ScopeEntry newEntry = new ScopeEntry(symbol, current);
        entries.put(name, newEntry);
    }
```
This is related to the issue https://github.com/ballerina-platform/ballerina-lang/issues/27816

Any idea?

**Comment by pubudu91** [comment]
This looks like a limitation in the way we have modeled the symbol space. Not sure how feasible it would be to change this at this point. :grimacing: 

Btw, wasn't there a map which kept a list of entries against a name, instead of a single entry, when we looked at the symbol resolver? Is this something at a lower level than that?

**Reply by dulajdilshan** [reply]
This is in the SymbolEnter, where we add entries to the scope with `defineSymbol` method.

**Reply by pubudu91** [reply]
Had a chat with @dulajdilshan about this. Not sure whether it'd be feasible to consider fixing this for alpha. Since this is a bug fix anyways, shall we defer this to post-alpha? @hasithaa @nadeeshaan

**Reply by nadeeshaan** [reply]
I think with the current time-line better approach would be to address this in post-alpha. In a related note, could we observe the same issue with imports namesapace as well?
